src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -258,13 +258,16 @@ private function addWorkflowSection(ArrayNodeDefinition $rootNode)
                                     ->end()
                                 ->end()
                                 ->validate()
-                                    ->always(function ($v) {
-                                        if (isset($v['type']) && isset($v['service'])) {
-                                            throw new \InvalidArgumentException('"type" and "service" could not be used together.');
-                                        }
-
-                                        return $v;
-                                    })
+                                    ->ifTrue(function ($v) { return !isset($v['type']) && !isset($v['service']); })
+                                    ->thenInvalid('One of "type" or "service" must be configured.')
+                                ->end()
+                                ->validate()
+                                    ->ifTrue(function ($v) { return isset($v['type']) && isset($v['service']); })
+                                    ->thenInvalid('"type" and "service" cannot be used together.')
+                                ->end()
+                                ->validate()
+                                    ->ifTrue(function ($v) { return isset($v['arguments']) && isset($v['service']); })
```

Comment:
I cannot confirm that. `requiresAtLeastOneElement()` works as expected when playing with the fixtures I added in this PR.